src/core/a-animation.js

```diff
@@ -373,6 +379,19 @@ function getAnimationValues (el, attribute, dataFrom, dataTo, currentValue) {
   };
 
   /**
+   * Match the scema type to color
+   * @return {bool} if the schema is of type color
+   */
+  function isColor () {
+    var componentName = attributeSplit[0];
+    var propertyName = attributeSplit[1];
+    var component = el.components[componentName];
+    var schema = component && component.schema;
+    if (schema) { return schema[propertyName].type === 'color'; }
```

Comment:
Now that I think about this. We could animate a component that it's not defined on the element yet (before the animation runs like)
````html
<a-entity geometry="..."><a-animation attribute="material.color" from="red" to="blue"></a-animation></a-entity>
````
Is this case covered? I think you would not be able to check ```el.components``` to retrieve the schema.